app/src/main/java/com/android/ubclaunchpad/driver/session/SessionInfoActivity.java

```diff
@@ -41,14 +48,31 @@ protected void onCreate(Bundle savedInstanceState) {
         final String sessionName = getIntent().getStringExtra(StringUtils.SESSION_NAME);
         TextView textViewSessionName = (TextView) findViewById(R.id.viewSessionName);
         textViewSessionName.setText(sessionName);
-
-        final DatabaseReference session = FirebaseUtils.getDatabase()
+        final Button goButton = (Button) findViewById(R.id.go_Button);
+        goButton.setVisibility(View.INVISIBLE);
+        session = FirebaseUtils.getDatabase()
                 .child(StringUtils.FirebaseSessionEndpoint)
                 .child(sessionName);
         final String UID = FirebaseUtils.getFirebaseUser().getUid();
         final User currentUser;
         try {
             currentUser = UserManager.getInstance().getUser();
+
+            session.addListenerForSingleValueEvent(new ValueEventListener() {
+                @Override
+                public void onDataChange(DataSnapshot dataSnapshot) {
+                    String uid = dataSnapshot.child("sessionHostUid").getValue(String.class);
+                    if (uid.equals(FirebaseUtils.getFirebaseUser().getUid())) {
+                        Toast.makeText(getBaseContext(), "You are the chosen one!", Toast.LENGTH_LONG).show();
+                        goButton.setVisibility(View.VISIBLE);
+                    }
+                }
+
+                @Override
+                public void onCancelled(DatabaseError databaseError) {
+                }
+            });
+
```

Comment:
I found this listener blocking the listener below so that we can no longer add users to sessions, and the "go" button doesn't show up. The code works when I move what's inside "onDataChange" to the second listener.